src/ui-test/rspServerProviderOperationsTest.ts

```diff
@@ -0,0 +1,100 @@
+import { ServersActivityBar } from "./server/ui/serversActivityBar";
+import { ServerState } from "./common/serverState";
+import { WebDriver, VSBrowser, NotificationType, Workbench, InputBox } from "vscode-extension-tester";
+import { RSPServerProvider } from "./server/ui/rspServerProvider";
+import { notificationExists, serverHasState } from "./server/serverUtils";
+import { expect } from 'chai'
+
+
+const ERROR_CREATE_NEW_SERVER = 'Unable to create the server';
+const ERROR_NO_RSP_PROVIDER = 'there are no rsp providers to choose from';
+const YES = 'Yes';
+const USE_FROM_DISK = 'No, use server on disk';
+
+/**
+ * @author Ondrej Dockal <odockal@redhat.com>
+ */
+export function rspServerProviderActionsTest() {
+    describe('Verify RSP Server provider actions', () => {
+
+        let driver: WebDriver;
+        let serverProvider: RSPServerProvider;
+        let serversActivityBar: ServersActivityBar;
+
+        before(() => {
+            driver = VSBrowser.instance.driver;
+            serversActivityBar = new ServersActivityBar();
+        });
+
+        beforeEach(async function() {
+            this.timeout(30000);
+            await serversActivityBar.open();
+            serverProvider = await serversActivityBar.getServerProvider();
+            const state = await serverProvider.getServerState();
+            if (state == ServerState.Unknown || state == ServerState.Starting)
+                try {
+                    await driver.wait(async () => { return await serverHasState(serverProvider, ServerState.Started);}, 10000 );
+                } catch (error) {
+                    console.log(error + ": Server was not started within 10 s on startup");
+                }
```

Comment:
The error message can be put directly into the wait: `driver.wait(condition, timeout, err_message)`

Wouldn't it be easier to let the error propagate? The tests require the provider to be started anyway.

Reply:
Thanks for the first suggestion.
Second, you are right, it was leftover for previous logic how to handle test env. at the start of every test. So throwing error is correct.